snippets/core/system-text-json/csharp/DeserializeNullToNonnullableType.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SystemTextJsonSamples
+{
+    public class DeserializeNullToNonnullableType
+    {
+        public static void Run()
+        {
+            string jsonString;
+            var wf = WeatherForecastFactories.CreateWeatherForecast();
+
+            var serializeOptions = new JsonSerializerOptions();
+            serializeOptions.WriteIndented = true;
+            serializeOptions.Converters.Add(new DateTimeOffsetNullHandlingConverter());
+            jsonString = JsonSerializer.Serialize(wf, serializeOptions);
+            Console.WriteLine($"JSON with valid Date:\n{jsonString}\n");
+
+            var deserializeOptions = new JsonSerializerOptions();
```

Comment:
I don't think it is necessary to create both serialization and deserialization specific options.

You can have one `options` in the sample and re-use for both sides. It doesn't look like the sample requires two.

Reply:
Same applies elsewhere. Only have different options if they need to be specific.

Reply:
In most cases they have to be different variables because they are separate snippets in the doc.  If I used the same options variable I wouldn't be able to declare the variable in the second snippet., which appears as an independent snippet in the doc. The alternative would be to have separate snippets to declare the options variable and use it, i.e., 3 instead of 2 snippets n the code file and 4 instead of 2 in the doc. I can do that but splitting what is really one snippet into two separate ones would look strange in the doc.  In the instances where that's not an issue I'm changing the code to use a single options variable.